src/btree/bt_compact.c

```diff
@@ -141,8 +141,9 @@ __wt_compact(WT_SESSION_IMPL *session)
 		 * read, set its generation to a low value so it is evicted
 		 * quickly.
 		 */
-		WT_ERR(__wt_tree_walk(session, &ref,
-		    WT_READ_COMPACT | WT_READ_NO_GEN | WT_READ_WONT_NEED));
+		WT_ERR(__wt_tree_walk_custom_skip(session, &ref,
```

Comment:
@keithbostic Please take a close look at this change - I'm not super familiar with the compact code, and tree walk is complex.